pennylane/labs/resource_estimation/resource_config.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""This module contains the base class for qubit management"""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+
+from pennylane.labs.resource_estimation.ops.qubit.parametric_ops_single_qubit import (
+    ResourceRX,
+    ResourceRY,
+    ResourceRZ,
+)
+from pennylane.labs.resource_estimation.resource_operator import ResourceOperator
+from pennylane.labs.resource_estimation.templates import (
+    ResourceSelectPauliRot,
+    ResourceQubitUnitary,
+    ResourceQROMStatePreparation,
+    ResourceMPSPrep,
+    ResourceAliasSampling,
+    ResourceSelectTHC,
+    ResourcePrepTHC,
+    ResourceQubitizeTHC,
+)
+
+
+class ResourceConfig:
+    r"""A container to track the configuration for errors, precisions, and custom decompositions for the
+    resource estimation pipeline.
+    """
+
+    def __init__(self) -> None:
+        self.conf = {
+            ResourceRX: {"error_rx": 1e-9},
+            ResourceRY: {"error_ry": 1e-9},
+            ResourceRZ: {"error_rz": 1e-9},
+            ResourceSelectPauliRot: {"precision": 1e-9},
+            ResourceQubitUnitary: {"precision": 1e-9},
+            ResourceQROMStatePreparation: {"precision": 1e-9},
+            ResourceMPSPrep: {"precision": 1e-9},
+            ResourceAliasSampling: {"precision": 1e-9},
+            ResourceQubitizeTHC: {"rotation_precision": 15, "coeff_precision": 15},
+            ResourceSelectTHC: {"rotation_precision": 15, "coeff_precision": 15},
+            ResourcePrepTHC: {"rotation_precision": 15, "coeff_precision": 15},
+        }
+        self._decomp_tracker = {}
```

Comment:
Maybe just call them `custom_decomps`? It says "tracker" but it isn't really "tracking" anything. It's simply a library of custom decompositions that you refer to.